### PR TITLE
Add working add item button, uuid.

### PR DESCRIPTION
**Pull request by JanekJaniak**

### CODE REVIEW

File: src/components/RenderList.vue [hunk 1]
```diff
@@ -1,34 +1,35 @@
 <template>
   <div>
-      <div v-for="item in items" :key="item.id">
+      <div v-for = "(item, index) in items" :key = "item.id">
```
Comment (jan-gajdowicz) [inline src/components/RenderList.vue]: not the right way to use spaces
in HTML use `v-for=""`

File: src/components/RenderList.vue [hunk 2]
```diff
@@ -1,34 +1,35 @@
...
-            <strong>Gross Value:{{calculateGrossValue(ingredient.quantity,ingredient.price,ingredient.vat)}}</strong>
+            <strong>Net Value:{{calculateNetValue(ingredient.quantity, ingredient.price)}}</strong>
+            <strong>Gross Value:{{calculateGrossValue(ingredient.quantity, ingredient.price, ingredient.vat)}}</strong>
           </div>
         </div>
         <div>
-          <input placeholder="Ingredient name" @input="handleInput">
-          <button @click="addIngredient">Add new ingredient</button>
+          <input placeholder = "Ingredient name" @keypress.enter = "addIngredient(index)"  @input = "handleInput">
+          <button @click = "addIngredient(index)" :itemIndex = index >Add new ingredient</button>
```
Comment (jan-gajdowicz) [inline src/components/RenderList.vue]: What's the use for `itemIndex`?

File: src/components/RenderList.vue [hunk 3]
```diff
@@ -77,17 +78,21 @@ export default {
       handleInput(event) {
         this.value = event.target.value;
       },
-      addIngredient() {
+      addIngredient(index) {
+        const newID = uuid.v4();
         const newIngredient = {
-          id:Date.now(),
+          id:newID,
```
Comment (jan-gajdowicz) [inline src/components/RenderList.vue]: in js object the spacing shoul be: `name: value`

File: src/components/RenderList.vue [hunk 4]
```diff
@@ -77,17 +78,21 @@ export default {
...
           quantity:"",
           price:"",
           vat:"1",
           gross:"",
           net:"",
         };
-        this.items[0].ingredients.push(newIngredient);
+        if(this.value) {
+        this.items[index].ingredients.push(newIngredient);
+        this.value = "";
```
Comment (jan-gajdowicz) [inline src/components/RenderList.vue]: Indent

File: src/components/RenderList.vue [hunk 5]
```diff
@@ -77,17 +78,21 @@ export default {
       handleInput(event) {
         this.value = event.target.value;
       },
-      addIngredient() {
+      addIngredient(index) {
```
Comment (jan-gajdowicz) [inline src/components/RenderList.vue:86]: Separate methods with an empty line

**Review by jan-gajdowicz — CHANGES_REQUESTED** [review]

File: src/components/RenderList.vue (hunk 1 above)
Comment (JanekJaniak) [reply src/components/RenderList.vue]: ✔️

File: src/components/RenderList.vue (hunk 2 above)
Comment (JanekJaniak) [reply src/components/RenderList.vue]: No use. Deleted.

File: src/components/RenderList.vue (hunk 3 above)
Comment (JanekJaniak) [reply src/components/RenderList.vue]: ✔️

File: src/components/RenderList.vue (hunk 5 above)
Comment (JanekJaniak) [reply src/components/RenderList.vue:86]: ✔️

File: src/components/RenderList.vue [hunk 6]
```diff
@@ -65,29 +66,39 @@ export default {
...
+          vat: "1",
+          gross: "",
+          net: "",
         };
-        this.items[0].ingredients.push(newIngredient);
+
+        if(this.value) {
+        this.items[index].ingredients.push(newIngredient);
+        this.value= "";
+        }
```
Comment (jan-gajdowicz) [inline src/components/RenderList.vue]: Missing indent

File: src/components/RenderList.vue [hunk 7]
```diff
@@ -65,29 +66,39 @@ export default {
               { text: "8", value:1.08},
               { text: "23", value:1.23}
       ],
+      value: "",
     }
   },
   methods: {
```
Comment (jan-gajdowicz) [inline src/components/RenderList.vue]: Separate with a new line

File: src/components/RenderList.vue [hunk 8]
```diff
@@ -65,29 +66,39 @@ export default {
               { text: "8", value:1.08},
               { text: "23", value:1.23}
```
Comment (jan-gajdowicz) [inline src/components/RenderList.vue]: Use single indent, as elsewhere

File: src/components/RenderList.vue [hunk 9]
```diff
@@ -1,34 +1,35 @@
...
         <div>
-          <div v-for="ingredient in item.ingredients" :key="ingredient.id">
+          <div v-for="(ingredient, index) in item.ingredients" :key="ingredient.id">
             <strong>{{ingredient.name}}</strong>
-            <label for = "weight">Weight:</label>  
-            <input id = "weight" placeholder="Edit" v-model.number="ingredient.quantity" type ="number" />
-            <label for = "price">Price/Kg:</label>
-            <input id = "price" placeholder="Edit" v-model.number="ingredient.price" type ="number" />
+            <label :for="`weight-${index}`" >Weight:</label>  
+            <input :id="`weight-${index}`" placeholder="Edit" v-model.number ="ingredient.quantity" type="number"/>
```
Comment (jan-gajdowicz) [inline src/components/RenderList.vue]: `v-model.number="` unnecessary space

File: src/components/RenderList.vue [hunk 10]
```diff
@@ -65,29 +66,39 @@ export default {
...
+          price: "",
+          vat: "1",
+          gross: "",
+          net: "",
         };
-        this.items[0].ingredients.push(newIngredient);
+
+        if(this.value) {
+        this.items[index].ingredients.push(newIngredient);
+        this.value= "";
```
Comment (jan-gajdowicz) [inline src/components/RenderList.vue]: `this.value = ""` space

**Review by jan-gajdowicz — CHANGES_REQUESTED** [review]

File: src/components/RenderList.vue (hunk 6 above)
Comment (JanekJaniak) [reply src/components/RenderList.vue]: ✔️

File: src/components/RenderList.vue (hunk 7 above)
Comment (JanekJaniak) [reply src/components/RenderList.vue]: ✔️

File: src/components/RenderList.vue (hunk 8 above)
Comment (JanekJaniak) [reply src/components/RenderList.vue]: ✔️

File: src/components/RenderList.vue (hunk 9 above)
Comment (JanekJaniak) [reply src/components/RenderList.vue]: ✔️

File: src/components/RenderList.vue (hunk 10 above)
Comment (JanekJaniak) [reply src/components/RenderList.vue]: ✔️

**Review by jan-gajdowicz — APPROVED** [review]
Excelent job!

**Pull request merged**